tests/document/indexes.py

```diff
@@ -792,6 +792,27 @@ class User(Document):
 
         User.drop_collection()
 
+    def test_unique_and_primary_create(self):
+        """Create a new record with a duplicate primary key
+        throws an exception
+        """
+
+        class User(Document):
+            name = StringField(primary_key=True)
+            password = StringField()
+
+        User.drop_collection()
+
+        User.objects.create(name='huangz', password='secret')
+        def duplicate_primary_key():
```

Comment:
Instead of defining an inner function, a more elegant solution is to use a `with` block like so:
```
with self.assertRaises(NotUniqueError):
    User.objects.create(name='huangz', password='secret2')
```
```